packages/core/src/components/non-ideal-state/nonIdealState.tsx

```diff
@@ -119,21 +105,30 @@ export class NonIdealState extends AbstractPureComponent<NonIdealStateProps> {
                         tabIndex={-1}
                     />
                 </div>
-            );
-        }
-    }
-
-    private maybeRenderText() {
-        const { description, title } = this.props;
-        if (title == null && description == null) {
-            return undefined;
-        } else {
-            return (
+            )
+        );
+    };
+
+    const maybeRenderText = () => {
+        return (
+            !!title &&
+            !!description && (
```

Comment:
I believe the logic here should be:
```ts
!!title || !!description
```
Otherwise, the text will only render only if _both_ title and description are set. Having just a title and no description (or vice versa) is a valid configuration for this component.

Better yet, can we be a bit more explicit with our operators here? (since this is TypeScript)
```ts
title != null || description != null
```